modules/aioseop_sitemap.php

```diff
@@ -4284,6 +4373,87 @@ public function get_all_post_priority_data( $include = 'any', $status = 'publish
 
 			$links = $this->get_prio_from_posts( $posts, $this->get_default_priority( 'post', true ), $this->get_default_frequency( 'post', true ) );
 			$links = array_merge( $links, $this->get_archive_prio_from_posts( $posts ) );
+
+			$is_sitemap_indexes_disabled = empty( $this->options['aiosp_sitemap_indexes'] );
+			if ( $is_sitemap_indexes_disabled || ( ! $is_sitemap_indexes_disabled && 'page' === $include ) ) {
+				$links = $this->get_prio_freq_static_homepage( $links );
+				$links = $this->update_woocommerce_shop_timestamp( $links );
+			}
+
+			return $links;
+		}
+
+		/**
+		 * The get_prio_freq_static_homepage() function.
+		 *
+		 * Sets the priority and frequency for the homepage if it is static.
+		 *
+		 * @since 3.2.0
+		 *
+		 * @param array $links
+		 * @return array $links
+		 */
+		private function get_prio_freq_static_homepage( $links ) {
+			if ( 0 === get_option( 'page_on_front' ) ) {
+				return $links;
+			}
+			$prio = $this->options['aiosp_sitemap_prio_homepage'];
+			$freq = $this->options['aiosp_sitemap_freq_homepage'];
+
+			$homepage_url = get_site_url() . '/';
+			$count = count( $links );
+			for ( $i = 0; $i < $count; $i++ ) {
+				if ( $homepage_url === $links[ $i ]['loc'] ) {
+					if ( 'no' !== $prio ) {
+						$links[ $i ]['priority'] = $prio;
+					}
+					if ( 'no' !== $freq ) {
+						$links[ $i ]['changefreq'] = $freq;
+					}
+				}
+			}
+			return $links;
+		}
+
+		/**
+		 * The update_woocommerce_shop_timestamp() function.
+		 *
+		 * Updates the Last Change timestamp for the WooCommerce shop page based on the last modified product - #2126.
+		 *
+		 * @since 3.2.0
+		 *
+		 * @param array $links
+		 * @return array $links
+		 */
+		private function update_woocommerce_shop_timestamp( $links ) {
+			if ( ! aioseop_is_woocommerce_active() ) {
+				return $links;
+			}
+
+			$shop_page_url = get_permalink( wc_get_page_id( 'shop' ) );
+			$count = count( $links );
+			for ( $i = 0; $i < $count; $i++ ) {
```

Comment:
This For loop can be reduced by using either

```
$test_key = array_search( $shop_page_url, array_column( $links, 'loc' ) );
$test_keys = array_keys( array_column( $links, 'loc' ), $shop_page_url );
```

Reply:
Done.